src/OptionSettlement.sol

```diff
@@ -185,17 +181,24 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
             optionInfo.exerciseAmount,
             optionInfo.underlyingAmount,
             optionInfo.exerciseTimestamp,
-            optionInfo.expiryTimestamp
+            optionInfo.expiryTimestamp,
+            optionInfo.nextClaimId
             );
     }
 
     /// @inheritdoc IOptionSettlementEngine
     function write(uint256 optionId, uint112 amount) external returns (uint256 claimId) {
-        if (tokenType[optionId] != Type.Option) {
+        if (amount == 0) {
+            revert AmountWrittenCannotBeZero();
+        }
+
+        (uint160 _optionId, uint96 claimIndex) = getDecodedIdComponents(optionId);
+        if (claimIndex != 0) {
+            // claim index must be zero in lower 96b
```

Comment:
This seems wrong. Multiple claims can be written against an option.

Reply:
Unless I am misunderstanding what you are checking here.

Reply:
Yep I agree with your first comment and think this code written here is aligned with that. 

What I was envisioning is that any tokenId with nonzero lower 96b corresponds to a Claim NFT. Any token ID with zero lower 96b corresponds to a fungible Option erc20. So here, you're writing an `amount` of `optionId` which will return a unique claim ID for that given option type. This check here is just ensuring that the lower 96b are zero, and the user did not supply a `claimId` instead of an `optionId`